tests/testthat/test-table_with_settings_ui.R

```diff
@@ -21,3 +23,127 @@ testthat::test_that("Table with settings: UI screenshots", {
+    app <- shinytest2::AppDriver$new(
+      app_tws(),
+      name = "tws",
+      variant = "app_tws_ui",
+    )
+
+    app$wait_for_idle(timeout = default_idle_timeout)
+
+    # Check if there are two buttons above the table.
+    table_buttons_selector <- "#table_with_settings-table-with-settings > div.table-settings-buttons"
+    table_buttons <-
+      app$get_html(table_buttons_selector) %>%
+      rvest::read_html() %>%
+      rvest::html_elements("button")
+    testthat::expect_length(table_buttons, 2)
+    # Check is the first one is a toggle button.
+    testthat::expect_equal(
+      table_buttons[[1]] %>%
+        rvest::html_attr("data-toggle"),
+      "dropdown"
+    )
+    # First button has specific font-awesome icon.
+    dwnl_button <- "#table_with_settings-downbutton-dwnl_state"
+    testthat::expect_equal(
+      app$get_html(dwnl_button) %>%
+        rvest::read_html() %>%
+        rvest::html_element("i") %>%
+        rvest::html_attr("class"),
+      "fas fa-download"
+    )
+
+    # Click the first TABLE button.
+    app$click(selector = dwnl_button)
+    app$wait_for_idle(timeout = default_idle_timeout)
+
+    # Review the content of the toggle.
+    testthat::expect_equal(
+      app$get_text("#table_with_settings-downbutton-file_format-label"),
+      "File type"
+    )
+
+    file_format_text <- app$get_text("#table_with_settings-downbutton-file_format > div")
+    testthat::expect_match(file_format_text, "formatted txt\n", fixed = TRUE)
+    testthat::expect_match(file_format_text, "csv\n", fixed = TRUE)
+    testthat::expect_match(file_format_text, "pdf\n", fixed = TRUE)
+
+    testthat::expect_equal(
+      app$get_text("#table_with_settings-downbutton-file_name-label"),
+      "File name (without extension)"
+    )
+
+    testthat::expect_match(
+      app$get_value(input = "table_with_settings-downbutton-file_name"),
+      sprintf("table_%s", gsub("-", "", Sys.Date()))
+    )
+
+    pagination <- "#dropdown-menu-table_with_settings-downbutton-dwnl .paginate-ui .form-group.shiny-input-container"
+    pagination_text <- app$get_text(pagination)
+    testthat::expect_match(pagination_text, "Paginate table:\n", fixed = TRUE)
+    testthat::expect_match(pagination_text, "lines / page\n", fixed = TRUE)
+
+    download_button <- app$get_html("#table_with_settings-downbutton-data_download > i") %>% rvest::read_html()
+    testthat::expect_equal(
+      download_button %>%
+        rvest::html_node("i") %>%
+        rvest::html_attr("class"),
+      "fas fa-download"
+    )
+    testthat::expect_equal(
+      download_button %>%
+        rvest::html_node("i") %>%
+        rvest::html_attr("aria-label"),
+      "download icon"
+    )
+
+    app$click(selector = "input[value='.csv']")
+    # check that pagination is missing
+    # app$get_text(pagination) # this returns values even though pagination is missing from the view
+    testthat::expect_false(
+      any(
+        unlist(
+          app$get_js(
```

Comment:
@averissimo since I am using just `shinytest2::AppDriver` I can not use `teal::TealAppDriver$is_visible()` method : /